### PR TITLE
refactor: use strings.Cut for strings parsing

### PR DESCRIPTION
**Pull request by bedakb**
This PR uses Go 1.18 `strings.Cut` function and replaces usage of `strings.Split` and `strings.SplitN` where possible.

Resolves https://github.com/zalando/skipper/issues/1963

### CODE REVIEW

File: filters/auth/oidc_introspection.go [hunk 1]
```diff
@@ -64,12 +64,12 @@ func (spec *oidcIntrospectionSpec) CreateFilter(args []interface{}) (filters.Fil
 	switch filter.typ {
 	case checkOIDCQueryClaims:
 		for _, arg := range sargs {
-			slice := strings.SplitN(arg, ":", 2)
-			if len(slice) != 2 {
+			path, queries, found := strings.Cut(arg, ":")
+			if !found || path == "" {
```
Comment (AlexanderYastrebov) [inline filters/auth/oidc_introspection.go:68]: The check for non-empty path might make sense but is out of scope for this refactoring. I suggest we keep existing logic here and elsewhere.

File: filters/builtin/compress.go [hunk 2]
```diff
@@ -13,6 +12,8 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/andybalholm/brotli"
```
Comment (AlexanderYastrebov) [inline filters/builtin/compress.go:15]: While not a big deal, this does not seem to be related to the refactoring.

File: tracing/tracers/basic/basic.go [hunk 3]
```diff
@@ -21,34 +21,34 @@ func InitTracer(opts []string) (opentracing.Tracer, error) {
...
-		switch parts[0] {
+		k, v, _ := strings.Cut(o, "=")
+		switch k {
 		case "drop-all-logs":
 			dropAllLogs = true
 
 		case "sample-modulo":
-			if len(parts) == 1 {
-				return nil, missingArg(parts[0])
+			if v == "" {
```
Comment (AlexanderYastrebov) [inline tracing/tracers/basic/basic.go:30]: Note: for equivalent refactoring it should check for `found` value as `v` could be empty, e.g. for `strings.Cut("foo=", "=")`.
Here and in most other places it is fine to check for empty `v` as empty value should not be permitted.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
@bedakb Thank you for your contribution.

File: tracing/tracers/basic/basic.go (hunk 3 above)
Comment (bedakb) [reply tracing/tracers/basic/basic.go:30]: I think `v == ""` must be kept regardless `found` is `true` or `false`. This particular example `strings.Cut("foo=", "=")` will return `"foo", "", true`.
Also It covers this case `strings.Cut("drop-all-logs", "=")` -> `"drop-all-logs", "", false`, so If we'd introduce a `found` check what should be do in case `found == false` ? Is doing a `continue` enough, so we skip to the next option ?

File: filters/builtin/compress.go (hunk 2 above)
Comment (bedakb) [reply filters/builtin/compress.go:15]: ah sorry, that's just my IDE sorting the imports

**Comment by szuecs** [comment]
:+1:
